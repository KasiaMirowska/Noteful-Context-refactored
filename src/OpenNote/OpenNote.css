.open-note {
    color: black;
    display: block;
}
.open-note2 {
    color:rgb(150, 74, 3);
}
.openNote {
    display: flex;
    margin-bottom: 0;
}
.date {
    color: black;
}

.content-container {
    display: block;
    background-color: rgb(158, 156, 154);
    padding: 30px;
    padding-bottom: 100px;
}
