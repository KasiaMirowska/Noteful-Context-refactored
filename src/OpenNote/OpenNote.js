import React from 'react';
import NotefulContext from '../NotefulContext';
import './OpenNote.css'



export default class OpenNote extends React.Component {
    static contextType = NotefulContext;
    render() {
        const note = this.context.notes.find(note => note.id === this.props.match.params.openNoteId);
    
    return (
        <div>
            <h2>Notes</h2>
            <div className='open-view'>
                <h3>{note.name}</h3>
                {note.modified}       
                <button className='delete-button'>Delete Note</button>
            </div> 
            <p className='content'>{note.content}</p>    
        </div>
                       
        
    );
    }
    
}