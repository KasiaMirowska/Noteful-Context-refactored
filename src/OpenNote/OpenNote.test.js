import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import OpenNote from './OpenNote';
import { BrowserRouter } from 'react-router-dom';



describe('OpenNote component', () => {
    it('OpenNote renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><OpenNote match={{params:{openNoteId : 'some id'}}}/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const openNote = renderer.create(<BrowserRouter><OpenNote match={{params:{openNoteId : 'some id'}}}/></BrowserRouter>);
        
        expect(openNote.toJSON()).toMatchSnapshot();
        
    });
});
