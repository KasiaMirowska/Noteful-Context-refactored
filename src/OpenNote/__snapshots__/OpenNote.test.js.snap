// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenNote component renders the UI as expected 1`] = `null`;
